lib/sql.js

```diff
@@ -680,6 +680,24 @@ SQLConnector.prototype.update = function(model, where, data, options, cb) {
 };
 
 /**
+ * Update all instances that match the where clause with the given data, but
+ * without returning the count.
+ * @param {String} model The model name
+ * @param {Object} where The where object
+ * @param {Object} data The property/value object representing changes
+ * to be made
+ * @param {Object} options The options object
+ * @param {Function} cb The callback function
+ */
+SQLConnector.prototype.updateNoCount = function(model, where, data,
+  options, cb) {
+  var stmt = this.buildUpdate(model, where, data, options);
+  this.execute(stmt.sql, stmt.params, options, function(err) {
```

Comment:
The issue is that the DB/2 update won't actually _give_ the count back by default. As a result, when we try to bake the update in as a nested query for a select to satisfy the original updateAll API requirement, we end up with the deadlock specified in the relevant PMR. 

Ignoring the count allows us to drop that information (since the query isn't going to give it to us anyway).